src/reducers/ast.js

```diff
@@ -124,6 +132,23 @@ export function hasSymbols(state: OuterState, source: Source): boolean {
   return !!state.ast.getIn(["symbols", source.id]);
 }
 
+export function isEmptyLineInSource(
+  state: OuterState,
+  line: number,
+  selectedSource: Source
+) {
+  const emptyLines = getEmptyLines(state, selectedSource);
+  return emptyLines.includes(line);
+}
+
+export function getEmptyLines(state: OuterState, source: Source) {
+  if (!source) {
```

Comment:
Looking at this again, we should probably return `[]` instead of `false`.

Reply:
(Prevents an error in `isEmptyLineInSource` because we immediately use `.includes`; I think it's reasonable to always expect an array from this function)

Reply:
good call